client/reader/components/quick-post/index.tsx

```diff
@@ -76,6 +77,16 @@ function QuickPost( {
 		}
 	}, [ postContent ] );
 
+	// Set initial selected site if none is selected
+	useEffect( () => {
+		if ( ! selectedSiteId ) {
+			const siteId = getMostRecentlySelectedSiteId( state ) || currentUser?.primary_blog;
+			if ( siteId ) {
+				dispatch( setSelectedSiteId( siteId ) );
+			}
+		}
+	}, [ selectedSiteId, dispatch, state, currentUser ] );
+
```

Comment:
Alternatively to handling this all with an effect hook and dispatching site selection changes, etc. We could simply leverage the site selectors `selectedSiteId` prop below (around line 193), and simply add the fallback (` selectedSiteId={ selectedSiteId || mostRecentlySelectedSiteId || primarySiteId || undefined }` - or similarly move that chaining to a single variable.

also `currentUser?.primary_blog` probably works fine, but we should have a selector for primary blog id as well if we want to be consistent.

Reply:
Note if we are updating it that way or using the shared variable for this, we likely have to make sure to use it consistently here - like in the wpcom request around L93 and your full editor link ~L202

Reply:
Thanks for the catch! I believe the latest updates take care of these issues.